dataplane/linux/int_dataplane.go

```diff
@@ -942,7 +942,7 @@ func ConfigureDefaultMTUs(hostMTU int, c *Config) {
 		c.VXLANMTU = hostMTU - vxlanMTUOverhead
 	}
 	if c.Wireguard.MTU == 0 {
-		if c.KubernetesProvider == config.ProviderAKS && c.RouteSource == "WorkloadIPs" {
+		if c.Wireguard.EncryptHostTraffic {
```

Comment:
This line should actually be:
`if c.KubernetesProvider == config.ProviderAKS &&c.Wireguard.EncryptHostTraffic {`
because we only need to tweak the MTU like this on AKS.